File: Northwind.Host.Web/Content/js/models/Customer.js
/**
	Modelo que representa un Customer

	@class		Customer
	@extends	Northwind.Model
	@namespace	Northwind
	@module		@Northwind
**/

Northwind.Customer = Northwind.Model.extend({
	companyName: DS.attr('string'),
	contactName: DS.attr('string'),
	contactTitle: DS.attr('string'),
	address: DS.attr('string'),
	city: DS.attr('string'),
	region: DS.attr('string'),
	postalCode: DS.attr('string'),
	country: DS.attr('string'),
	phone: DS.attr('string'),
	fax: DS.attr('string'),

	completeAddress: function () {

		return this.get('address') + '. ' + this.get('city') + '. ' + this.get('postalCode') + '. ' + this.get('country');

	}.property('address', 'city', 'postalCode', 'country'),
	    
    orders: DS.hasMany('order')
});

Northwind.Customer.reopen({

	required: ['contactName', 'companyName']

});

File: Northwind.ServiceModel/Dto/Customer.cs
using System.Collections.Generic;
using Northwind.ServiceBase;
using Northwind.ServiceBase.Relations;
using ServiceStack;

namespace Northwind.ServiceModel.Dto
{
	/// <summary>
	/// Clase que representa una entidad <see cref="Customer"/>
	/// </summary>	
	[Api("Create a single Customer.")]
	[Route("/customers", "POST")]
	[Route("/customers/{Id}", "PUT DELETE")]       
	public class Customer : CommonDto, IReturnVoid
	{		
		public string Id { get; set; }		
		public string CompanyName { get; set; }
		public string ContactName { get; set; }
		public string ContactTitle { get; set; }
		public string Address { get; set; }
		public string City { get; set; }
		public string Region { get; set; }
		public string PostalCode { get; set; }
		public string Country { get; set; }
		public string Phone { get; set; }
		public string Fax { get; set; }

		[Relation(RelationType.HasMany, typeof(Order))]
		public List<Order> Orders { get; set; }		
	}
}
